Sources/MapboxCoreNavigation/NavigationSettings.swift

```diff
@@ -115,6 +116,19 @@ public class NavigationSettings {
         state.utilizeSensorData
     }
     
+    /**
+     Defines approximate navigator prediction between location ticks.
+     
+     Due to discrete location updates, Navigator always operates data "in the past" so it has to make prediction about user's current real position. This interval controls how far ahead Navigator will try to predict user location.
+     
+     If not specified (`nilled`), default value will be used.
+     
+     You can override this property by using `NavigationSettings.initialize(directions:tileStoreConfiguration:navigationRouterType:alternativeRouteDetectionStrategy:utilizeSensorData:navigatorPredictionInterval:)` method.
+     */
+    public var navigatorPredictionInterval: TimeInterval? {
+        state.navigatorPredictionInterval
+    }
```

Comment:
A one-shot option with a read-only property isn’t really much of a setting (though I realize this isn’t the first read-only property in NavigationSettings). Would there be a use case for allowing the interval to be changed on the fly?

Reply:
We would need to do a refactor down to NN layer if we need such case. Starting with inability to modify basically any Navigator property after init.